erpnext/payroll/doctype/overtime_slip/overtime_slip.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe import _, bold
+from frappe.utils import get_datetime, getdate, get_link_to_form, formatdate
+from erpnext.payroll.doctype.payroll_entry.payroll_entry import get_start_end_dates
+from erpnext.payroll.doctype.gratuity.gratuity import get_salary_structure
+from frappe.model.document import Document
+class OvertimeSlip(Document):
+	def validate(self):
+		if not (self.from_date or self.to_date or self.payroll_frequency):
+			date = self.from_date or self.posting_date
+			data = get_frequency_and_dates(self.employee, date)
+
+			self.from_date = data[0].start_date
+			self.to_date = data[0].end_date
+			self.payroll_frequency = data[1]
+
+		self.validate_overlap()
+		if self.from_date >= self.to_date:
+			frappe.throw(_("From date can not be greater than To date"))
+
+		if not len(self.overtime_details):
+			self.get_emp_and_overtime_details()
+
+	def validate_overlap(self):
+		if not self.name:
+			# hack! if name is null, it could cause problems with !=
+			self.name = "new-overtime-slip-1"
```

Comment:
This should be handled properly. Fix in the framework? What problems does it cause?

Reply:
done